### ISSUE TITLE
Signs on roads can be edited

### ISSUE THREAD

**Issue by InfectedCatBite** [IntellectualSites/PlotSquared#4100]
### Server Implementation

Paper

### Server Version

1.20.1

### Describe the bug

Regular players can edit signs on roads.

### To Reproduce

1. Have an admin put a sign on a road.
2. Have a regular player right click on the sign and edit it.

### Expected behaviour

Players shouldn't be able to edit signs on roads without a flag to enable it.

Related, you might want to add a new plot (and road?) flag to allow editing. Right now, it's tied to the `use` flag. A separate flag would allow players to interact with signs (e.g., for another plugin to do stuff when the player clicks on the sign) but prevent them from editing the signs.

### Screenshots / Videos

![image](https://github.com/IntellectualSites/PlotSquared/assets/93386304/07b3ffb4-a96a-4108-bfd4-494a70fcec7c)

### Plot Debugpaste

Details below. Just try it yourself on your own setup.

### PlotSquared Version

PlotSquared version 7.0.0-SNAPSHOT commit af32399

### Checklist

- [X] I have included a Plot debugpaste.
- [X] I am using the newest build from https://www.spigotmc.org/resources/77506/ and the issue still persists.

### Anything else?

```
[P2] Area flags: [chat;true, block-ignition;true, use;#doors,#trapdoors,#buttons,#pressure_plates,#fence_gates,lever,tripwire_hook, vehicle-cap;20, vehicle-use;true]
[P2] Road flags: [redstone;true]
```

**Comment by NotMyFault** [comment]
> Have a regular player right click on the sign and edit it.

That behavior is guarded by `plots.admin.interact.road`. Players are unable to modify signs on the road without granting explicit permissions.
I was able to verify my previous sentences after testing the setup with the steps described in the issue.
Due to the lack of a debugpaste, I'm unable to take a more profound look.

**Comment by InfectedCatBite** [comment]
Yes, you are correct regarding that permission. I give players that permission so they can use buttons on roads. I suppose the road use flag could do the same thing. I'll try that instead.

Thank you and I hope you'll consider having a separate flag for editing signs.
